# Sentiment Analysis - Project Group 6

## Importing file data into lists from #JobsReport(tweets), Positive Words and Negative Words

### Importing tweets from #JobsReport (tweet .txt file)

In [361]:
# Open the file containing tweets '#JobsReport.txt' in read mode.
f = open('#JobsReport.txt','r', encoding = 'utf8', errors = 'ignore')

In [362]:
# Read the file opened above in a variable 'tweets'
tweets = f.read()

In [363]:
#tweets in list and converting them in lower case.
tweet_list = [w.strip(' ,.!?:;') for w in tweets.lower().split()] 

In [364]:
# Closing the file.
f.close()

### Importing words from Negative Words (.txt file)

In [365]:
# Open the file named 'Negative_Words_Final.txt' which contains the negative words in read mode.
n = open('Negative_Words_Final.txt','r', encoding = 'utf8', errors = 'ignore')

In [366]:
# Read the file opened above in a variable 'neg'
neg = n.read()

In [367]:
neg = neg.strip(' ,.!?:;').lower().split() #negative words in a list

In [368]:
n.close()

### Importing tweets from Positive Words (.txt file)

In [369]:
# Open the file named 'Positive_Words.txt' which contains the positive words in read mode.
p = open('Positive_Words.txt','r', encoding = 'utf8', errors = 'ignore')

In [370]:
# Read the file opened above in a variable 'pos'
pos = p.read()

In [371]:
pos = pos.strip(' –,').replace(',', '').lower().split() #positive words in a list

In [372]:
p.close()

### Check if all files are closed

In [373]:
# Check if the files are closed.
f.closed

True

In [374]:
n.closed

True

In [375]:
p.closed

True

### Code to calculate general sentiment

In [376]:
#Initializing variables to calculate the general sentiment.
count_neg = 0
count_pos = 0
total = 0
percent_pos = 0
percent_neg = 0

In [377]:
#Counting the words in the tweet list that occur in posiitve words list or negative words list.
#Positive words have been displayed as positive values & negative words as negative values.
for word in tweet_list:
    if word in neg:
        count_neg-=1
        total-=1
    elif word in pos:
        count_pos+=1
        total+=1

In [378]:
#total depicts the general feeling of the tweets (positive or negative)
#count_neg holds the no of positive words in the tweets
#count_pos holds the no of negative words in the tweets
print('Count of Positive Words:', count_pos)
print('Count of Negative Words:', count_neg)
print('Count of positive & negative words:', count_pos - count_neg)
print('General Sentiment:', total)

Count of Positive Words: 119493
Count of Negative Words: -46107
Count of positive & negative words: 165600
General Sentiment: 73386


### After running sentiment analysis the general sentiment count came out to be 73386 (positive) by subtracting the total positive words found in the tweets 119493 and the total negative words found in the tweets 46107. The total number of words in the tweets are 655605 and the total of positive & negative words is 165600. The remaining words are neutral and contribute nothing towards the Sentiment. The general sentiment is positive that means public have positive sentiments to the Jobs Report released by US President Donald Trump.


<i> To find out the strength of positive sentiment, we have taken the percentage of positive words in the total of positive and negative words.</i>
##### Criteria: If the General Sentiment is positive then the overall sentiment is positive. If the percent of positive/negative words is greater than or equal to 50% then the sentiment is STRONGLY POSITIVE/NEGATIVE.
<i>Neutral word count has not been included as they have no impact on the total and therefore can be ignored.</i>

In [384]:
if total > 0:
    percent_pos = (count_pos/(count_pos - count_neg)) * 100
    print('Percent of postitive words:', percent_pos, '%')
    if percent_pos >= 50:
        print ("The general sentiment of Jobs Report in US is STRONGLY POSITIVE.")
    else:
        print ("The general sentiment of Jobs Report in US is WEAKLY POSITIVE.")
elif total < 0:
    percent_neg = (count_neg/(count_pos - count_neg)) * 100
    print('Percent of negative words:', percent_neg, '%')
    if percent_neg >= 50:
        print ("The general sentiment of Jobs Report in US is STRONGLY NEGATIVE.")
    else:
        print ("The general sentiment of Jobs Report in US is WEAKLY NEGATIVE.")
else:
    print ("The general sentiment of Jobs Report in US is NEUTRAL.")

Percent of postitive words: 72.15760869565217 %
The general sentiment of Jobs Report in US is STRONGLY POSITIVE.


### 72% being on the higher side shows that the sentiment is STRONGLY POSITIVE.